json_theme/lib/src/codec/theme_encoder.dart

```diff
@@ -7114,7 +7114,7 @@ class ThemeEncoder {
         'buttonTheme': encodeButtonThemeData(value.buttonTheme),
         'canvasColor': encodeColor(value.canvasColor),
         'cardColor': encodeColor(value.cardColor),
-        'cardTheme': encodeCardTheme(value.cardTheme),
```

Comment:
I think we could make in non breaking by adding a condition like this before:

```
      Object cardTheme = value.cardTheme;
      if (cardTheme is CardTheme) {
        cardTheme = cardTheme.data;
      } else if (cardTheme is! CardThemeData) {
        throw ArgumentError('cardTheme must be either a CardThemeData or a CardTheme');
      }
```

And then something like `encodeCardTheme(cardTheme as CardThemeData)`

Like done from flutter themselves: https://github.com/flutter/flutter/pull/155129/files#diff-9d9b537da8ee2b1c3d29615749e838767109e585bd59cdbd4ac356e30f344505R504

Reply:
@barbl058  would that be an option?

Reply:
Sorry, forget what I said, there's no way of handling this, as the classes I check don't even yet exist.